### PR TITLE
Add Sentry Releases

### PR DESCRIPTION
**Pull request by jmhobbs**
Adds sourcemaps and releases to Sentry:

- Has `tsup` build with sourcemaps
- Uses tsup environment variable substitution on `SENTRY_RELEASE` to add to `init`
- Uses `@sentry/cli` to upload sourcemaps and create the release
- Removes the sourcemaps so they don't get bundled into the distribution
- Continues with the `auto` release flow

<details>
  <summary>📦 Published PR as canary version: <code>11.12.1--canary.1084.11238719287.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@11.12.1--canary.1084.11238719287.0
  # or 
  yarn add chromatic@11.12.1--canary.1084.11238719287.0
  ```
</details>

### CODE REVIEW

File: vitest.config.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export default defineConfig({
     exclude: [...configDefaults.exclude, '**/getParentCommits.test.ts'],
     coverage: {
       provider: 'v8',
-      exclude: ['vitest.no-threads.config.ts', ...coverageConfigDefaults.exclude],
+      exclude: ['vitest.no-threads.config.ts', 'scripts/**', ...coverageConfigDefaults.exclude],
```
Comment (jmhobbs) [inline vitest.config.ts:9]: No reason vitest should be looking at these right now.

File: .github/workflows/package-size-main.yml [hunk 2]
```diff
@@ -22,6 +22,10 @@ jobs:
         run: yarn install --immutable
       - name: build
         run: yarn build
+      - name: Remove Sourcemaps
+        run: |
+          rm -f dist/*.map
+          rm -f action/*.map
```
Comment (codykaup) [inline .github/workflows/package-size-main.yml:28]: Should we use the `yarn clean:sourcemaps` script here?
```suggestion
          yarn clean:sourcemaps
```

File: .github/workflows/package-size.yml [hunk 3]
```diff
@@ -24,6 +24,10 @@ jobs:
         run: yarn install --immutable
       - name: build
         run: yarn build
+      - name: Remove Sourcemaps
+        run: |
+          rm -f dist/*.map
+          rm -f action/*.map
```
Comment (codykaup) [inline .github/workflows/package-size.yml:30]: Same here
```suggestion
          yarn clean:sourcemaps
```

**Review by codykaup — APPROVED** [review]

**Comment by jmhobbs** [comment]
Merging over `lint-and-test` as that PR is still open and we are covered by the CircleCI version.

**Pull request merged**